WorldCat Discovery.js

```diff
@@ -0,0 +1,185 @@
+		  translator.setString(responseDocument);
+		  translator.setHandler('itemDone', function() {
+			var item;
+			item = arguments[arguments.length - 1];
+			item.extra = void 0;
+			item.archive = void 0;
+			if (item.libraryCatalog === 'http://worldcat.org') {
+			  item.libraryCatalog = 'Open WorldCat';
+			}
+			item.title = item.title.replace(/\s+:/, ':');
+			return item.complete();
+		  });
+		  return translator.getTranslatorObject(function(translator) {
+			translator.options.defaultItemType = 'book';
+			translator.options.typeMap = {
+			  'ELEC': 'book'
+			};
+			return translator.doImport();
+		  });
+		}));
+	  }
+	  return _results;
+	};
+	items = {};
+	if (isActionRecord) {
+	  activeRecord = doc.getElementById('main');
+	  oclcNum = activeRecord.getAttribute('data-oclcnum');
+	  items[oclcNum] = parseOpenURLData(activeRecord).title;
+	  return saveItems(items);
+	} else {
+	  if (activeRecord) {
+		oclcNum = activeRecord.getAttribute('data-oclcnum');
+		items[oclcNum] = parseOpenURLData(activeRecord).title;
+		return saveItems(items);
+	  } else {
+		for (_i = 0, _len = records.length; _i < _len; _i++) {
+		  record = records[_i];
+		  oclcNum = record.getAttribute('data-oclcnum');
+		  items[oclcNum] = parseOpenURLData(record).title;
+		}
+		return Zotero.selectItems(items, saveItems);
+	  }
+	}
+  }
+};
+
+pageIsActionRecord = function(url) {
+  return url.indexOf('/oclc/') !== -1 && url.indexOf('/search?') === -1;
+};
+
+parseOpenURLData = function(baseItemElement) {
+  var contextObject, item, itemType;
+  contextObject = Zotero.Utilities.xpathText(baseItemElement, './/span[@class="Z3988"]/@title');
+  item = new Zotero.Item();
+  Zotero.Utilities.parseContextObject(contextObject, item);
+  itemType = Zotero.Utilities.xpathText(baseItemElement, './/div[@class="basicbib"]//i[contains(@class, "icon")]/@class');
+  item.type = itemType.indexOf('icon-snd') !== -1 ? 'audioRecording' : itemType.indexOf('icon-compfile') !== -1 || itemType.indexOf('icon-intmm') !== -1 ? 'computerProgram' : itemType.indexOf('icon-map') !== -1 ? 'map' : item.itemType;
+  return item;
+};
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "http://goshen.on.worldcat.org/search?sortKey=LIBRARY_PLUS_RELEVANCE&databaseList=638&queryString=monkeys&changedFacet=format&scope=&format=all&format=Compfile&subformat=Compfile%3A%3Acompfile_digital&database=all&author=all&year=all&yearFrom=&yearTo=&language=all",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "http://goshen.on.worldcat.org/oclc/9323519",
+		"items": [
+			{
+				"itemType": "conferencePaper",
+				"title": "Microcomputers in K-12 education: second annual conference proceedings",
+				"creators": [
+					{
+						"lastName": "Conference of Microcomputers in K-12 Education",
+						"creatorType": "author",
+						"fieldMode": 1
+					},
+					{
+						"lastName": "Barrette",
+						"firstName": "Pierre.",
```

Comment:
This looks like it needs a cleaning. However, AFAIK we shouldn't delete points if only initials are present.